pkgs/development/go-modules/generic/default.nix

```diff
@@ -105,16 +107,18 @@ let
     inherit (go) GOOS GOARCH;
 
     GO111MODULE = "on";
+    GOFLAGS = "-mod=vendor";
 
     configurePhase = args.configurePhase or ''
       runHook preConfigure
 
       export GOCACHE=$TMPDIR/go-cache
       export GOPATH="$TMPDIR/go"
       export GOSUMDB=off
-      export GOPROXY=file://${go-modules}
-
+      export GOPROXY=off
       cd "$modRoot"
+      rm -rf vendor
+      cp -r ${go-modules} vendor
```

Comment:
Would `ln -s ${go-modules} vendor` work?

Reply:
Applied it and it works :)